### PR TITLE
Lists: fix pagination edge cases and add Swagger docs

### PR DESCRIPTION
**Pull request by Julia-Dubenchuk**
Addressed correctness issues in the lists pagination response and improved API visibility in Swagger.

Checklist:

- [x] Fix hasPreviousPage/hasNextPage returning misleading true when total = 0 — both flags now guard on total > 0
- [x] Add edge-case test for total = 0, page > 1 in lists.service.spec.ts
- [x] Fix controller spec: mockReturnValue → mockResolvedValue + await, to reflect the real async service contract
- [x] Decorate PaginatedListsResponseDto and its meta shape with @ApiProperty so Swagger renders the full response schema
- [x] Extract PaginationMetaDto into src/common/dto/ for reuse across future paginated endpoints

### CODE REVIEW

File: src/lists/dto/paginated-lists-response.dto.ts [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-  data: List[];
-  meta: {
-    page: number;
-    limit: number;
-    total: number;
-    totalPages: number;
-    hasNextPage: boolean;
-    hasPreviousPage: boolean;
-  };
+  @ApiProperty({ type: [List] })
```
Comment (organizm) [inline src/lists/dto/paginated-lists-response.dto.ts:6]: Мінор: `@ApiProperty({ type: [List] })` посилається на TypeORM entity без `@ApiProperty` на полях — у Swagger UI `data` може показуватися як `{}`. Варіанти: окремий `ListResponseDto` або `@ApiProperty` на потрібних полях entity. Не блокує merge.

File: src/lists/test/lists.controller.spec.ts [hunk 2]
```diff
@@ -72,7 +72,7 @@ describe('ListsController', () => {
       expect(mockListsService.findAll).toHaveBeenCalledWith(user.sub, 1, 10);
```
Comment (organizm) [inline src/lists/test/lists.controller.spec.ts]: Мінор: фікстура досі з `user.sub` — у #22 вже `user.id`. На реальному JWT `request.user` це `User` без `sub`; якщо мержити окремо від #22, варто мати на увазі.

**Review by organizm — APPROVED** [review]
Точковий і коректний PR: pagination flags, async spec, Swagger. Блокерів немає — approve. Повний фікс `user.sub` → `user.id` — у #22 цього ланцюжка.

### Похвально
- Коректний фікс `hasNextPage` / `hasPreviousPage` при `total = 0` — обидва `false`, навіть якщо `page > 1`.
- Тест на edge case `total = 0, page = 2` у `lists.service.spec.ts`.
- `mockReturnValue` → `mockResolvedValue` + `await` у controller spec — відповідає async-контракту сервісу.
- Винесення `PaginationMetaDto` у `src/common/dto/` — готово до reuse на інших paginated endpoints.

### Перевірено й закрито
- Логіка `totalPages = Math.ceil(total / limit)` при `total = 0` дає `0` — узгоджено з новими прапорцями.
- `PaginationMetaDto` покриває всі поля колишнього inline `meta` object.
- Зміна не чіпає query/DB — лише response meta + документація.
